### PR TITLE
Room endpoints

### PR DESCRIPTION
**Pull request by mark-marshall**
# Description

Please include a summary of the change and a link to which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
These are the ROOM endpoints. They allow an array of hotel rooms to be POSTed and retrieved, room name changes and deletions. POST requests are done as an array to allow for multiple room uploads at a time. roomRoutes are a separate file from hotelRoutes for cleanliness.

These endpoints have note-form documentation and need to be run through the APIDoc template, as well as tested.

Fixes # (issue)
#11 

## Change status

- [ ] Complete, tested, ready to review and merge

- [x] Complete, but not tested (may need new tests)

- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?
Manually via Postman and via MongoDB Compass.

Please describe steps taken to ensure this feature is functional and does not break other features:
Separate routes file with new routes.

# Checklist:

- [x] My code follows the style guidelines of this project

- [x] I have performed a self-review of my own code

- [x] I have commented my code, particularly in hard-to-understand areas

- [ ] I have made corresponding changes to the documentation, if relevant

- [ ] I have run the app using my feature and ensured that no functionality is broken

- [x] My changes generate no new warnings

- [x] There are no merge conflicts

### CODE REVIEW

**Review by ConConRob — COMMENTED** [review]
Looks really good. The utils and middle ware keeps everything clean.

File: middleware/validateRoomObjectId.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const errorMessage = require('../utils/errorMessage');
+
+const validateRoomObjectId = (req, res, next) => {
+  const { _roomId } = req.params;
```
Comment (Pav0l) [inline middleware/validateRoomObjectId.js]: Please update the function in `validateObjectId.js` to:
`const _id = req.params._roomId || req.params._id;` 
so we can reuse that one and keep only one function for ID validation.

File: middleware/validateRoomsArr.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  if (!Array.isArray(roomsArr)) {
+    res.status(400).json(errorMessage.typeArr);
+  } else if (roomsArr.filter(room => !room.name).length) {
+    res.status(400).json(errorMessage.noRoomName);
+  } else {
+    next();
+  }
+};
+
+module.exports = validateRoomsArr;
```
Comment (Pav0l) [inline middleware/validateRoomsArr.js:15]: This could be validated in `hotelSchema` in `hotelModels.js`, with `type: Array`.

**Review by Pav0l — CHANGES_REQUESTED** [review]
As always, awesome job Mark! 🔥 

Couple of comments below

File: middleware/validateRoomObjectId.js (hunk 1 above)
Comment (mark-marshall) [reply middleware/validateRoomObjectId.js]: Changing this to generalised subDoc function as discussed.

File: middleware/validateRoomsArr.js (hunk 2 above)
Comment (mark-marshall) [reply middleware/validateRoomsArr.js:15]: I think that if I don't have the type check here, and someone sends a non-array,  my code will break way before it gets to the db validation -> was a way of handling the error from that essentially.

Comment (Pav0l) [reply middleware/validateRoomsArr.js:15]: 👍

Comment (Pav0l) [reply middleware/validateRoomsArr.js:15]: Validation-ception 👍

**Review by Pav0l — APPROVED** [review]

**Pull request merged**